src/demo/DemoTestLoanService.java

```diff
@@ -340,6 +341,43 @@ private static void testEditLoanAmountForActiveRevolvingCreditLoans() throws Mam
 					"The loan account does not meet the prerequisites, therefore its loan amount will not be updated.");
 		}
 	}
+	
+	private static void testEditPrincipalPaymentAmountForActiveRevolvingCreditLoans() throws MambuApiException {
+
+		methodName = new Object() {}.getClass().getEnclosingMethod().getName();
+		System.out.println("\nIn " + methodName);
+		
+		LoanProductType loanProductType = demoProduct.getLoanProductType();
+		LoansService loanService = MambuAPIFactory.getLoanService();
+		AccountState accountState = loanService.getLoanAccount(NEW_LOAN_ACCOUNT_ID).getAccountState();
+
+		// Edit loan's principal payment amount for active revolving credit loan
+		if (loanProductType.equals(LoanProductType.REVOLVING_CREDIT) && (accountState.equals(AccountState.ACTIVE) ||
+				(accountState.equals(AccountState.ACTIVE_IN_ARREARS)))) {
```

Comment:
It's ok, but there is an extra set of parentheses here. I think we can delete those that are around the very last inner condition; the ones that start right before the second occurrence of accountState.equals(...).